[23기 박태환] [Mission 3,4] Add: 메인페이지 레이아웃 / 기능구현 완료 및 셀프 리팩토링 진행중

mekemeke94 opened the pull request:
- 로그인 페이지 기본 레이아웃 작성
- 로그인 input tag 유효성 검사 로직 작성 (id, pw 한 글자 이상 입력하면
  로그인버튼 활성화)

>wecode PR message template 및 체크 리스트 입니다. 
아래 사항들을 전부 작성/체크 하시고 PR 해주세요!

## 수정 사항 간략한 한줄 요약
(여기에 수정 사항에 대한 간략한 한줄 요약/제목 작성해 주세요.)
7/12(월)
-로그인 레이아웃 작업 완료
7/13(화)
-로그인 인풋 태그 로직 완성
7/15(목)
-메인 페이지 레이아웃 및 기능구현 완료
7/16(금)
-셀프 리팩토링 진행중

## 수정 사항들 자세한 내용
-로그인창 레이아웃을 완성했습니다. 현재 인스타그램 창과는 다르지만 tobe폴더에 있는 예시 로그인 창과 최대한 비슷하게 구현하려고 했습니다.
-로그인 input창에 id 입력란과 pw 입력란에 각각 한글자 이상씩 입력하면 로그인 버튼이 활성화 되면서 버튼 색이 바뀌도록 구현했습니다.

-메인 페이지 레이아웃 완성했습니다. 
- 댓글입력 시 댓글창에 입력되도록 기능 구현 완료했습니다.
- 댓글 창에 새로 입력될 시 간격 조절이 필요한 것 같아 전체적인 레이아웃도 다시 한 번 수정했습니다.

-셀프 리팩토링 진행중입니다.
일단 bottom-up형식으로 만들지 않아 하나하나 확인하며 수정했습니다.
css도 레이아웃에 영향을 많이 주는 속성 순서에 따라 다시 재배열 했습니다.

## 기타 질문 및 특이 사항
아이디 인풋과 패스워드 인풋에 각각 클래스 이름으로 addEventListener로 함수를 적용시켰는데 혹시 다른 방법은 없는지 궁금합니다.

셀프리팩토링 진행중입니다. css코드 수를 줄이고 싶은데 레이아웃 구성 시 원하는 대로 구성하기 위해 div를 너무 많이 생성한 것 같아 css코드를 줄이는데 어려움이 있습니다. 어떤 식으로 줄여야 할 지 궁급합니다.


## 체크 리스트 (아래 사항들이 전부 체크되어야만 merge가 됩니다!)
- [x] 필요한 test들을 완료하였고 기능이 제대로 실행되는지 확인 하였습니다.
- [x] Wecode의 코드 스타일 가이드에 맞추어 코드를 작성 하였습니다.
- [x] 제가 의도한 파일들과 수정 사항들만 커밋이 된 것을 확인 하였습니다.
- [x] 본 수정 사항들을 팀원들과 사전에 상의하였고 팀원들 모두 해당 PR에 대하여 알고 있습니다.

joonsikyang commented on students/10th/yoonjiyoung/js/login.js:
```diff
@@ -0,0 +1,37 @@
+const loginBtn = document.querySelector(".login__btn");
+const loginForm = document.querySelector(".login__input");
+
+const PAINT_BTN = "login__paintBtn";
```
이 변수가 특정 함수 안에서만 사용된다면 그 함수 안에서 정의해도 좋을 것 같네요! 변수 사용한 점은 Good🤘

joonsikyang commented on students/10th/yoonjiyoung/js/login.js:
```diff
@@ -0,0 +1,37 @@
+    : loginBtn.classList.remove(PAINT_BTN);
+}
+
+function handleBtn(event) {
```
- 함수 명으로 paintBtn, handleBtn 이렇게 작성해주셨는데 좀 더 명확하면 좋을 것 같습니다!
- 이런경우라면 paintBtn >>> activateBtn , handleBtn >>> moveToMain 이런식으로 기능의 의미가 들어간 이름이 더 좋겠네요.

yojuyoon replied:
수정사항 반영하였습니다!

joonsikyang commented on students/10th/yoonjiyoung/js/login.js:
```diff
@@ -0,0 +1,37 @@
+function handleBtn(event) {
+  event.preventDefault();
+  location.href = "file:///Users/yoonjiyoung/Downloads/westagram/main.html?";
+  console.log(event);
+  console.log(location);
```
확인 끝난 console.log는 삭제해주세요!

joonsikyang commented on students/10th/yoonjiyoung/js/login.js:
```diff
@@ -0,0 +1,37 @@
+  console.log(location);
+}
+
+loginId.addEventListener("keyup", paintBtn);
+loginPw.addEventListener("keyup", paintBtn);
```
두 줄은 한 줄로 바꿀 수 있을 것 같습니다. 각각의 요소에 같은 이벤트와 같은 함수가 실행되는데 어떻게 하면 하나로 합칠 수 있을지 고민해보시기 바랍니다 :)

yojuyoon replied:
수정사항 반영하였습니다!

joonsikyang commented on students/10th/yoonjiyoung/js/login.js:
```diff
@@ -0,0 +1,37 @@
+
+loginId.addEventListener("keyup", paintBtn);
+loginPw.addEventListener("keyup", paintBtn);
+loginForm.addEventListener("submit", handleBtn);
```
페이지 이동 함수까지👍

joonsikyang commented on students/10th/yoonjiyoung/js/main.js:
```diff
@@ -0,0 +1,90 @@
+
+let comment = [];
+
+function init() {
```
지금 모든 함수가 init에 묶여 있어서 함수를 분리해도 좋을 것 같습니다. 그럼 PR template에 있는 질문 사항도 어느정도 해결될 수 있을 것 같아요!

yojuyoon replied:
PR template 질문 수정사항 Fix 커밋에 반영하여 해결하였습니다..! 때문에 이번 리뷰반영 push에 함수를 분리하지 않고 커밋하였습니다!

joonsikyang commented on students/10th/yoonjiyoung/js/main.js:
```diff
@@ -0,0 +1,90 @@
+const navLogo = document.querySelector(".nav__logo-img");
+const navSearch = document.querySelector(".nav__search");
+const navSearchVal = document.querySelector(".nav__search-input");
```
해당 변수는 요소 자체를 담고 있어서 Val이 들어간 이름이 아주 살짝 어색한 것 같습니다.=

joonsikyang commented on students/10th/yoonjiyoung/js/main.js:
```diff
@@ -0,0 +1,90 @@
+          const delBtn = document.createElement("button");
+          delBtn.classList.add("delBtn");
+          delBtn.innerText = "X";
+          delBtn.addEventListener("click", () => {
+            li.parentNode.removeChild(li);
+          });
```
Arrow Function  활용 Good!

joonsikyang commented on students/10th/yoonjiyoung/js/main.js:
```diff
@@ -0,0 +1,90 @@
+    window.location.reload();
+  });
+}
+init();
```
전체적으로 기능 구현도 깔끔하고, arrow function, forEach 메소드 등 배운 것들이 잘 적용된 것 같네요!! 너무 좋습니다👍

joonsikyang commented on students/10th/yoonjiyoung/login.html:
```diff
@@ -0,0 +1,32 @@
+  <head>
+    <meta charset="utf-8" />
+    <meta name="viewport" content="width=device-width" />
+    <title>repl.it</title>
```
HTML 복습 겸 title 내용도 좀 더 멋진걸로 바꿔보세요 :)

joonsikyang commented on students/10th/yoonjiyoung/style/login.css:
```diff
@@ -0,0 +1,74 @@
+  box-sizing: border-box;
+  font-family: "Arial";
+}
+a {
```
한 칸 띄워주세요!

joonsikyang replied:
아래 한 번씩 검토하면서 붙여서 작성되어 있는 부분 전부 띄워주시기 바랍니다.

yojuyoon replied:
수정사항 반영하였습니다!

joonsikyang commented on students/10th/yoonjiyoung/style/main.css:
```diff
@@ -0,0 +1,387 @@
+  position: absolute;
+  right: 0px;
+  bottom: 5px;
+}
```
- 반응형까지..🤘
- 전체적으로 css가 깔끔하고 class명에서도 고민의 흔적이 많이 느껴지는 것 같습니다.
- 요소와 요소 사이에 한 줄 띄는 것만 신경써주세요. 가독성이 좋아집니다.

yojuyoon replied:
감사합니다🙏

mekemeke94 committed:
Add: 로그인 페이지 레이아웃 / 기능 완료
- 로그인 페이지 기본 레이아웃 작성
- 로그인 input tag 유효성 검사 로직 작성 (id, pw 한 글자 이상 입력하면
  로그인버튼 활성화)

mekemeke94 committed:
Add: 메인페이지 레이아웃 및 댓글 입력 기능 구현 완료
- 메인페이지 기본 레이아웃 작성
- 댓글 창에 한 글자 이상 들어올 시 게시 버튼 활성화 기능 구현
- 댓글 창에 문자 입력 후 게시버튼 클릭 시 댓글 추가 기능 구현
- 댓글 창에 문자 입력 후 엔터키 입력 시 댓글 추가 기능 구현

mekemeke94 committed:
Add: 셀프 리팩토링

- bottom - up 형식으로 전환
기존 height값으로 높이를 설정했던 것들을 컨텐츠 크기에 맞게 늘어날 수
있도록 불필요한 height 값을 제거.

- css 속성 순서 변경
css레이아웃에 영향을 많이 주는 속성 순으로 변경 후 가독성이 좋도록 한
줄씩 띄어줌.
